httpx/_transports/default.py

```diff
@@ -91,21 +163,44 @@ def __exit__(
     ) -> None:
         self._pool.__exit__(exc_type, exc_value, traceback)
 
-    def request(
+    def handle_request(
         self,
         method: bytes,
-        url: URL,
-        headers: Headers = None,
-        stream: httpcore.SyncByteStream = None,
-        ext: dict = None,
-    ) -> typing.Tuple[int, Headers, httpcore.SyncByteStream, dict]:
-        return self._pool.request(method, url, headers=headers, stream=stream, ext=ext)
+        url: typing.Tuple[bytes, bytes, typing.Optional[int], bytes],
+        headers: typing.List[typing.Tuple[bytes, bytes]],
+        stream: typing.Iterator[bytes],
+        extensions: dict,
+    ) -> typing.Tuple[
+        int, typing.List[typing.Tuple[bytes, bytes]], typing.Iterator[bytes], dict
+    ]:
+        with map_httpcore_exceptions():
+            status_code, headers, byte_stream, extensions = self._pool.request(
+                method=method,
+                url=url,
+                headers=headers,
+                stream=stream,  # type: ignore
```

Comment:
Do we / how do we plan to deal with this `type: ignore`? Shall we wrap in `httpcore.SyncIteratorStream` in the interim?

Reply:
Good call. Using the concrete `httpcore.IteratorByteStream()` and `httpcore.AsyncIteratorByteStream()` classes to wrap this in the correct interface.